ocaml/xapi-storage-script/main.ml

```diff
@@ -49,6 +49,23 @@ let backend_backtrace_error name args backtrace =
 let missing_uri () =
   backend_error "MISSING_URI" ["Please include a URI in the device-config"]
 
+(** return a unique 'domain' string for Dom0, so that we can plug disks
+  multiple times (e.g. for copy).
+
+  XAPI should give us a unique 'dp' (datapath) string, e.g. a UUID for storage migration,
+  or vbd/domid/device.
+  For regular guests keep the domain as passed by XAPI (an integer).
+ *)
+let domain_of ~dp ~vm' =
```

Comment:
Requiring a prime in named label is a bit awkward - is there a reason this is not just `~vm`?

Reply:
All the parameters where this function is called were called `vm'`, however renaming should be easy, I've added a commit at the beginning of this PR that renames `vm'` to `vm`, and then we can avoid having `'` in the labeled argument.